### PR TITLE
[MPR] spec: Add stubs, document parameter types, stub-enable 2 functions

### PR DESCRIPTION
**Pull request by SergeGautherie**
All that we learned so far.

JIRA issue: [CORE-14376](https://jira.reactos.org/browse/CORE-14376)

--

NT4 Winfile, example debug log:
```
WARNING: calling stub WNetGetDirectoryTypeW('D:\',0x00D1FF3C,0x1)
WARNING: calling stub WNetGetDirectoryTypeW('D:\',0x0012D4E4,0x0)
WARNING: calling stub WNetGetDirectoryTypeW('D:\efi',0x0012D4E4,0x0)
WARNING: calling stub WNetGetDirectoryTypeW('D:\loader',0x0012D4E4,0x0)
WARNING: calling stub WNetGetDirectoryTypeW('D:\reactos',0x0012D4E4,0x0)
```

### CODE REVIEW

File: dll/win32/mpr/mpr.spec [hunk 1]
```diff
@@ -56,38 +58,51 @@
...
+@ stub WNetFormatNetworkNameW # (wstr wstr ptr ptr long long)
 @ stdcall WNetGetCachedPassword(ptr long ptr ptr long)
+@ stub WNetGetConnection2A
+@ stub WNetGetConnection2W # (wstr ptr ptr)
+@ stub WNetGetConnection3A
+@ stub WNetGetConnection3W
 @ stdcall WNetGetConnectionA(str ptr ptr)
 @ stdcall WNetGetConnectionW(wstr ptr ptr)
-@ stub WNetGetDirectoryTypeA
+@ stdcall -stub WNetGetDirectoryTypeA(str ptr long)
```
Comment (SergeGautherie) [inline dll/win32/mpr/mpr.spec:87]: Stub-enabling `WNetGetDirectoryTypeA()` was suggested, but does not seem required.
Tell me if you prefer not to.

File: dll/win32/mpr/mpr.spec [hunk 2]
```diff
@@ -56,38 +58,51 @@
...
+@ stub WNetDirectoryNotifyW # (ptr wstr long)
 @ stdcall WNetDisconnectDialog1A(ptr)
 @ stdcall WNetDisconnectDialog1W(ptr)
 @ stdcall WNetDisconnectDialog(long long)
+@ stub WNetDisconnectDialog2 # (ptr long wstr long)
 @ stdcall WNetEnumCachedPasswords(str long long ptr long)
-@ stdcall WNetEnumResourceA(long ptr ptr ptr)
-@ stdcall WNetEnumResourceW(long ptr ptr ptr)
+@ stdcall WNetEnumResourceA(ptr ptr ptr ptr)
+@ stdcall WNetEnumResourceW(ptr ptr ptr ptr)
```
Comment (learn-more) [inline dll/win32/mpr/mpr.spec:74]: This is not the first time someone tells you that `long` is perfectly fine for a handle..

Comment (SergeGautherie) [reply dll/win32/mpr/mpr.spec]: Removed.

Comment (SergeGautherie) [reply dll/win32/mpr/mpr.spec]: PS: Although I still think being explicit/consistent is better.

**Pull request merged**